extensions/wikia/Helios/User.class.php

```diff
@@ -107,17 +107,16 @@ public static function newFromToken( \WebRequest $request ) {
 					// the code is borrowed from SpecialUserlogin
 					// @see PLATFORM-1261
 					if ( session_id() == '' ) {
-						$sessionId = substr(hash('sha256',$token),0,32);
-						wfSetupSession($sessionId);
+						$sessionId = substr( hash( 'sha256', $token ), 0, 32 );
+						wfSetupSession( $sessionId );
 						WikiaLogger::instance()->debug( __METHOD__ . '::startSession' );
 
-						// Update mTouched on user when he starts new MW session
-						// @see SOC-1326
-						$invalidateCacheThrottleTime = $request->getSessionData( self::INVALIDATE_CACHE_THROTTLE_SESSION_KEY );
-						if ( $invalidateCacheThrottleTime === null || $invalidateCacheThrottleTime < time() ) {
-							$request->setSessionData( self::INVALIDATE_CACHE_THROTTLE_SESSION_KEY, time() + self::INVALIDATE_CACHE_THROTTLE );
+						$lockKey = wfSharedMemcKey( 'login-from-helios', 'user-invalidate-cache', 'user', $user->getId() );
+						\WikiaDataAccess::criticalSection( $lockKey, 1000/*ms*/, function () use ( $user ) {
```

Comment:
@kvas-damian Lock is per user. Moreover I assume when lock cannot be acquired it means that some other request is doing the same. In such case IMHO we can just pass since the user_touched will be bumped by another process.